### PR TITLE
Disable proof parallelism in fiat-crypto.

### PR DESCRIPTION
**Pull request by ppedrot**

### CODE REVIEW

**Comment by ppedrot** [comment]
Still not enough :/. @JasonGross do you have other ideas?

File: .gitlab-ci.yml [hunk 1]
```diff
@@ -641,6 +641,8 @@ library:ci-fcsl_pcm:
 
 library:ci-fiat_crypto:
   extends: .ci-template-flambda
+  variables:
+    COQEXTRAFLAGS: "-async-proofs off"
```
Comment (SkySkimmer) [inline .gitlab-ci.yml:645]: Does that work? In ci-bedrock2.sh we have `COQMF_ARGS='-arg "-async-proofs-tac-j 1"'`

**Comment by JasonGross** [comment]
I've created https://github.com/mit-plv/fiat-crypto/issues/1512, it seems that file recently (sometime between Oct 23 and Nov 11) jumped from taking about 1 GB RAM to taking about 4 GB RAM

**Comment by ppedrot** [comment]
Can we deactivate some targets in the Coq CI while the problem is sorted out?

**Comment by JasonGross** [comment]
Try setting `SKIP_BEDROCK2=1` for the second invitation of `make` (all-except-compiled)?

**Comment by JasonGross** [comment]
Actually, I think we'll be able to solve this on our side today, so no need to make that change, I think. See https://github.com/mit-plv/fiat-crypto/issues/1512#issuecomment-1315226674

**Comment by JasonGross** [comment]
Can we try this on top of #16809?  Also, is there a difference between `-async-proofs-tac-j 1` and `-async-proofs off`?

**Comment by ppedrot** [comment]
Let's see if fiat-crypto goes through...

**Comment by ppedrot** [comment]
Still broken.

**Comment by ppedrot** [comment]
Regardless of any option `-async-proofs off` and `-async-proofs-j 1`, coqc still spawns a coqworker process. This is probably the root of the problem, with the first option set we should only have a single Coq process ever.

**Comment by JasonGross** [comment]
> Regardless of any option `-async-proofs off` and `-async-proofs-j 1`, coqc still spawns a coqworker process. This is probably the root of the problem, with the first option set we should only have a single Coq process ever.

It seems a bit unfortunate to lose the only(?) real CI test of par.  (Is par tested in the test-suit, btw?)

**Comment by JasonGross** [comment]
> Still broken.

Is it broken in the same way though? (How do I find the logs?). It seems that Fiat Crypto's own CI is now broken on master with a not_found anomaly that was introduced into Coq master in the past couple of days...

**Comment by ppedrot** [comment]
It was breaking [there](https://gitlab.com/coq/coq/-/jobs/3337463612). 
```
COQC src/Curves/Weierstrass/AffineProofs.v
Finished transaction in 1.242 secs (1.198u,0.043s) (successful)
Command terminated by signal 9
src/Curves/Weierstrass/AffineProofs.vo (real: 151.17, user: 4.44, sys: 0.81, mem: 1502628 ko)
Makefile.coq:792: recipe for target 'src/Curves/Weierstrass/AffineProofs.vo' failed
```

> not_found anomaly that was introduced into Coq master

:expressionless:

**Comment by ppedrot** [comment]
The Not_found anomaly is caused by a module issue, so it's likely to have been introduced by #16794.

**Comment by ppedrot** [comment]
I locally reverted #16794, let's see.

**Comment by Zimmi48** [comment]
IIUC this was working with a hack, but this does not work with the proper solution implemented in #16837 because the latter is broken. Can we restore the hack and merge? Having a green CI is more important than not using hacks in the CI configuration.

**Comment by ppedrot** [comment]
@Zimmi48 I've restored the hacky variant. If you want you can merge it as-is, since anyways fiat-crypto won't compile right now due a missing overlay with the last PR merged on master.

**Review by Zimmi48 — APPROVED** [review]
This was already tested previously.

**Comment by Zimmi48** [comment]
@coqbot merge now

**Comment by Zimmi48** [comment]
@JasonGross @coq/ci-maintainers This PR fixed fiat-crypto but it made it also dangerously close to the 3h timeout. I've two examples where it succeeded. The [first time](https://gitlab.com/coq/coq/-/jobs/3341117260), it took 161 minutes. The [second time](https://gitlab.com/coq/coq/-/jobs/3377006988), it took 175 minutes. And when trying to get a full pipeline to backport this PR and a few others, it timed out: https://gitlab.com/coq/coq/-/jobs/3380528344. What should we do, increase the timeout or build less of fiat-crypto?

**Comment by JasonGross** [comment]
Increasing the timeout would be nice.  We should also be able to build most of Fiat Crypto with -j2, but we'll have to do a bit of juggling of the targets on our side, I think.

**Comment by Zimmi48** [comment]
OK, I'll appreciate a PR bumping the fiat-crypto timeout to something reasonable for the foreseeable future. (FWIW, most of the time it seems to complete in 180 minutes. I had two other recent successes on the `v8.17` branch. But having a timeout that is too strict is a recipe for wasting build time when a job has to be re-run.)

**Comment by JasonGross** [comment]
The timeout is for preventing infinite loops, more or less, right?  Could we find the longest time that a single file takes to complete, double that, and set that as a per-file timeout?  (I'm guessing that'll be around 20 minutes, since I think we have one file that's 10 minutes or something.)  And then set the overall timeout to, let's say, 5 hours?  I hope we won't go much over 3,

**Comment by Zimmi48** [comment]
Well, I'd rather not take that much margin if never come close to this figure.

**Pull request merged**